amy/workshops/views.py

```diff
@@ -286,7 +286,22 @@ def get_context_data(self, **kwargs):
 
         is_usersocialauth_in_sync = len(self.object.github_usersocialauth) > 0
         context["is_usersocialauth_in_sync"] = is_usersocialauth_in_sync
+        consents = (
+            Consent.objects.filter(person=self.object)
+            .active()
+            .select_related("term", "term_option")
+        )
 
+        context["consents"] = {
+            "May contact": consents.filter(term__slug="may-contact")[0],
+            "Consent to publish profile": consents.filter(term__slug="public-profile")[
+                0
+            ],
+            "Consent to include name when publishing lessons": consents.filter(
+                term__slug="may-publish-name"
+            )[0],
+            "Privacy policy agreement": consents.filter(term__slug="privacy-policy")[0],
```

Comment:
This PR looks good, except for the additional queries the `context["consents"]` introduces - can you perhaps improve to no more than 1 query?